### PR TITLE
Bug 1885319: Revise DataVolumeTemplate API field

### PR DESCRIPTION
**Pull request by yaacov**
The vm.Spec.DataVolumeTemplate api list actually contains DataVolumes rather than a struct that restricts the fields to only the metadata and spec. This gives the user the impression they can set things like the DataVolume group/kind and status when they in fact cannot.

Refs: 
https://github.com/kubevirt/kubevirt/pull/4261
https://github.com/kubevirt/kubevirt/pull/4319

Screeshots:
The Bug (using patch 4216):
screenshot-localhost_9000-2020 10 06-09_34_54

Signed-off-by: yaacov <kobi.zamir@gmail.com>

### CODE REVIEW

**Comment by yaacov** [comment]
/hold freeze :snowboarder: 

cc:// @davidvossel

**Comment by jelkosz** [comment]
/hold cancel
the master is now opened for 4.7

File: frontend/packages/kubevirt-plugin/src/components/modals/clone-vm-modal/configuration-summary.tsx [hunk 1]
```diff
@@ -42,11 +43,13 @@ const getDisksDescription = (
     const volume = volumes.find((v) => v.name === disk.name);
     if (volume) {
       if (volume.dataVolume) {
-        let dataVolume = dataVolumeTemplates.find((dv) => getName(dv) === volume.dataVolume.name);
+        let dataVolume = dataVolumeTemplates.find(
+          (dv) => getName(dv) === volume.dataVolume.name,
+        ) as V1alpha1DataVolume;
```
Comment (rawagner) [inline frontend/packages/kubevirt-plugin/src/components/modals/clone-vm-modal/configuration-summary.tsx:48]: you dont have to typecast here and anywhere else. Just specify the correct types from the beginning

**Review by rawagner — CHANGES_REQUESTED** [review]

File: frontend/packages/kubevirt-plugin/src/components/modals/clone-vm-modal/configuration-summary.tsx (hunk 1 above)
Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/modals/clone-vm-modal/configuration-summary.tsx]: fixed :+1:

File: frontend/packages/kubevirt-plugin/src/k8s/wrapper/vm/vm-wrapper.ts [hunk 2]
```diff
@@ -221,7 +230,9 @@ export class VMWrapper extends K8sResourceWrapper<VMKind, VMWrapper> implements
       storages.map((storage) => storage.disk),
     );
     this.data.spec.template.spec.volumes = _.compact(storages.map((storage) => storage.volume));
-    this.data.spec.dataVolumeTemplates = _.compact(storages.map((storage) => storage.dataVolume));
+    this.data.spec.dataVolumeTemplates = _.compact(
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/k8s/wrapper/vm/vm-wrapper.ts:233]: we should be consistent with this approach here - pleas update `prependStorage` as well

File: frontend/packages/kubevirt-plugin/src/k8s/wrapper/vm/vm-wrapper.ts [hunk 3]
```diff
@@ -33,6 +34,14 @@ import { VirtualMachineImportModel, VirtualMachineModel } from '../../../models'
 import { buildOwnerReferenceForModel } from '../../../utils';
 import { transformDevices } from '../../../selectors/vm/devices';
 
+export const getDataVolumeTemplateSpec = (dv: V1alpha1DataVolume): V1DataVolumeTemplateSpec =>
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/k8s/wrapper/vm/vm-wrapper.ts:37]: what about 
```suggestion
export const convertToDataVolumeTemplateSpec = (dv: V1alpha1DataVolume): V1DataVolumeTemplateSpec =>
```

or something similar to this?

Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/k8s/wrapper/vm/vm-wrapper.ts:37]: also it probably shouldn't reside in vm-wrapper

**Review by suomiy — CHANGES_REQUESTED** [review]
also what about 
- tests ?
- clone-vm ?

File: frontend/packages/kubevirt-plugin/src/k8s/wrapper/vm/vm-wrapper.ts (hunk 2 above)
Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/k8s/wrapper/vm/vm-wrapper.ts:225]: done

File: frontend/packages/kubevirt-plugin/src/k8s/helpers/vm-clone.ts [hunk 4]
```diff
@@ -102,15 +102,29 @@ export class VMClone {
...
-          const clonedDVTemplate = new DataVolumeTemplate({
-            name: generateDataVolumeName(name, volume.name),
-            pvcSourceName: pvcName,
-            pvcSourceNamespace: this.oldVMNamespace,
-            accessModes: getPvcAccessModes(pvc),
-            volumeMode: getPvcVolumeMode(pvc),
-            size: getPvcStorageSize(pvc),
-            storageClassName: getPvcStorageClassName(pvc),
-          }).build();
+          const clonedDVTemplate: V1DataVolumeTemplateSpec = {
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/k8s/helpers/vm-clone.ts:105]: can we rather use a DataVolumeWrapper here for easier use?

We could add a special method there similar to asResource for compatibility with V1DataVolumeTemplateSpec

File: frontend/packages/kubevirt-plugin/src/k8s/helpers/vm-clone.ts [hunk 5]
```diff
@@ -134,15 +148,29 @@ export class VMClone {
...
-          const clonedDVTemplate = new DataVolumeTemplate({
-            name: generateDataVolumeName(name, volume.name),
-            pvcSourceName: dvName,
-            pvcSourceNamespace: this.oldVMNamespace,
-            accessModes: getDataVolumeAccessModes(dataVolume),
-            volumeMode: getDataVolumeVolumeMode(dataVolume),
-            size: getDataVolumeStorageSize(dataVolume),
-            storageClassName: getDataVolumeStorageClassName(dataVolume),
-          }).build();
+          const clonedDVTemplate: V1DataVolumeTemplateSpec = {
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/k8s/helpers/vm-clone.ts:151]: similar here

**Review by suomiy — CHANGES_REQUESTED** [review]

**Comment by yaacov** [comment]
force push to skip buggy CI,

@metalice please re-review

**Pull request merged**